api/v1alpha1/stage_types.go

```diff
@@ -220,6 +220,9 @@ type StageSpec struct {
 	// kargo.akuity.io/shard label with the value of this field. When this field
 	// is empty, the webhook will ensure that label is absent.
 	Shard string `json:"shard,omitempty" protobuf:"bytes,4,opt,name=shard"`
+	// Vars is a list of variables that can be referenced by the PromotionTemplate
+	// and the arguments of the Verification.
```

Comment:
Non-blocking nit. If we want to drive home the notion of "scopes" for the vars, it might help (and may be more future-proof, to boot) if we say these vars can be referenced _anywhere in the Stage spec that supports expressions_ and cite PromotionTemplate and Verification as good examples.